src/toolchain.rs

```diff
@@ -107,7 +108,10 @@ impl<'a> Toolchain<'a> {
         let path = cfg.toolchain_path(&name);
         if !Toolchain::exists(cfg, &name)? {
             return Err(match name {
-                LocalToolchainName::Named(name) => RustupError::ToolchainNotInstalled { name },
+                LocalToolchainName::Named(name) => {
+                    let is_active = matches!(cfg.active_toolchain(), Ok(Some((t, _))) if t == name);
```

Comment:
The following secretly calls `Toolchain::new()`, causing an infinite recursion.

https://github.com/rust-lang/rustup/blob/671b874c943e1f23f69adcf7b906fb1fe1077f27/src/config.rs#L704-L706

Reply:
Addressed by manually inlining the call and removing unnecessary data dependencies.